lib/foreman/telemetry.rb

```diff
@@ -109,15 +119,29 @@ def add_histogram(name, description, instance_labels = [], buckets = DEFAULT_BUC
     end
 
     def increment_counter(name, value = 1, tags = {})
+      return unless allowed?(tags)
       @sinks.each { |x| x.increment_counter("#{prefix}_#{name}", value, tags) }
     end
 
     def set_gauge(name, value, tags = {})
+      return unless allowed?(tags)
       @sinks.each { |x| x.set_gauge("#{prefix}_#{name}", value, tags) }
     end
 
     def observe_histogram(name, value, tags = {})
+      return unless allowed?(tags)
       @sinks.each { |x| x.observe_histogram("#{prefix}_#{name}", value, tags) }
     end
+
+    private
+
+    def allowed?(tags)
+      result = true
+      tags.each do |label, value|
+        regexp = @allowed_tags[label]
```

Comment:
Should we do this? Or is something else running that method? Given it's memoized, I'd prefer calling a method instead of variable.
```suggestion
        regexp = allowed_tags[label]
```